rust/src/measurements/tulap/make_tulap.tex

```diff
@@ -0,0 +1,154 @@
+\documentclass{article}
+\input{../../lib.sty}
+
+\usepackage[
+        backend=biber,
+        style=authoryear-comp,
+        sorting=nyt,
+    ]{biblatex}
+\addbibresource{ref.bib} 
+
+\usepackage{thmtools}
+\usepackage{thm-restate}
+\newtheorem{thm}{Theorem}
+\newtheorem{defn}{Definition} 
+\newtheorem{prop}{Proposition}
+
+\title{\texttt{fn make\_tulap}}
+\author{Yu-Ju Ku, Jordan Awan, Aishwarya Ramasethu, Michael Shoemate}
+\begin{document}
+
+\maketitle
+
+\contrib
+
+Proves soundness of \rustdoc{measurements/fn}{make\_tulap}.
+% in \asOfCommit{mod.rs}{f5bb719}.
+\texttt{make\_tulap} accepts parameters \texttt{input\_domain} and \texttt{input\_metric} specifying the input metric space.
+This consists of the space of non-null floating-point numbers, where distance is computed via the absolute distance.
+
+Additionally, the privacy parameters \texttt{epsilon} (of type \texttt{float}) and \texttt{delta} are fixed the privacy loss.
+The measurement assumes (but does not require) that input data (\texttt{arg}) follows the Binomial Distribution,
+and returns a sample from the Tulap distribution centered at \texttt{arg}.
+
+\section{Hoare Triple}
+
+\subsection*{Preconditions}
+\subsubsection*{Compiler-verified}
+\begin{itemize}
+    \item Argument \texttt{input\_domain} is of type \texttt{AtomDomain<f64>}.
+    \item Argument \texttt{input\_metric} is of type \texttt{AbsoluteDistance<f64>}.
+    \item Argument \texttt{epsilon} is of type \texttt{float}
+    \item Argument \texttt{delta} is of type \texttt{float}
+\end{itemize}
+
+\subsubsection*{Human-verified}
+None
+
+\subsection*{Pseudocode}
+
+\lstinputlisting[language=Python,firstline=2,escapechar=|]{./pseudocode/make_tulap.py}
+
+\subsection*{Postcondition}
+
+\validMeasurement{\texttt{(input\_domain, input\_metric, epsilon, \\ delta)}}{\texttt{make\_tulap}}
+
+\section{Proof}
+
+\begin{proof} 
+\textbf{(Privacy guarantee.)} 
+
+\rustdoc{traits/samplers/struct}{PartialSample}\texttt{.value}, hereafter referred to as just \texttt{value} (a function) 
+can only fail when the pseudorandom byte generator used in its implementation fails due to lack of system entropy. 
+This is usually related to the computer's physical environment and not the dataset. 
+The rest of this proof is conditioned on the assumption that \texttt{value} does not raise an exception. \\
+
+A canonical noise distribution (CND)~\parencite{awan2023canonical}, 
+captures whether a real-valued distribution is perfectly tailored to satisfy . 
+We formalize this in Definition~\ref{def2}~\parencite{awan2023canonical}". The definition uses $T$ which is the tradeoff function of $type 1$ and $type 2$ errors between two distributions. A larger $T$ would mean it is harder to distiguish between two distributions.  
+\begin{defn}\label{def2}  % Def 3.1
```

Comment:
\begin{defn}[\cite[Definition~3.1]{awan2023canonical}]
\end{defn}

Just use this.